packages/amplify-cli/src/extensions/amplify-helpers/update-tracked-files.ts

```diff
@@ -0,0 +1,80 @@
+import * as path from 'path';
+import fs from 'fs-extra';
+import {
+  $TSAny, JSONUtilities, pathManager, stateManager,
+} from 'amplify-cli-core';
+
+const {
+  readJson,
+} = JSONUtilities;
+
+/**
+ * Updates Cognito files that are tracked so that the diff is detected for an `amplify push`
+ */
+export const updateCognitoTrackedFiles = async (): Promise<void> => {
+  const currentCloudBackendDir = pathManager.getCurrentCloudBackendDirPath();
+  const localBackendDir = pathManager.getBackendDirPath();
+  const amplifyMeta = stateManager.getMeta();
+  const cognitoResource = stateManager.getResourceFromMeta(amplifyMeta, 'auth', 'Cognito', undefined, false);
+
+  if (!fs.existsSync(currentCloudBackendDir) || !cognitoResource) {
+    return;
+  }
+
+  const { resourceName } = cognitoResource;
+
+  if (await detectCognitoDiff(currentCloudBackendDir, localBackendDir, resourceName)) {
+    await addExtraLineToCliInputsJson(localBackendDir, resourceName);
+  }
+};
+
+const detectCognitoDiff = async (
+  currentCloudBackendDir: string,
+  localBackendDir: string,
+  resourceName: string,
+): Promise<boolean> => detectCognitoAttributesRequireVerificationBeforeUpdateDiff(
+  currentCloudBackendDir,
+  localBackendDir,
+  resourceName,
+);
+
+const detectCognitoAttributesRequireVerificationBeforeUpdateDiff = async (
+  currentCloudBackendDir: string,
+  localBackendDir: string,
+  resourceName: string,
+): Promise<boolean> => {
+  const cloudBackendUserAttrUpdateSettings = await readCfnTemplateUserAttributeSettings(currentCloudBackendDir, resourceName);
+  const backendUserAttrUpdateSettings = await readCfnTemplateUserAttributeSettings(localBackendDir, resourceName);
+  const updateNotInCloudBackend: boolean = !cloudBackendUserAttrUpdateSettings?.AttributesRequireVerificationBeforeUpdate
+    || cloudBackendUserAttrUpdateSettings?.AttributesRequireVerificationBeforeUpdate[0] !== 'email';
+  const updateInLocalBackend: boolean = backendUserAttrUpdateSettings?.AttributesRequireVerificationBeforeUpdate.length === 1
+    && backendUserAttrUpdateSettings?.AttributesRequireVerificationBeforeUpdate[0] === 'email';
+
+  return updateNotInCloudBackend && updateInLocalBackend;
+};
+
+type UserAttributeUpdateSettings = {
+  AttributesRequireVerificationBeforeUpdate: string[]
+}
+
+const readCfnTemplateUserAttributeSettings = async (
```

Comment:
the utility of this function is very localized to the **detectCognitoAttributesRequireVerificationBeforeUpdateDiff** function, so it may be worth declaring it as an arrow function within that; it won't see re-use outside of there. Also, resourceName is not changing, just the backendDir path; so you can exclude it from params in your new arrow function as the param will be in the same closure